webapp/src/main/java/com/box/l10n/mojito/rest/scheduledjob/ScheduledJobWS.java

```diff
@@ -0,0 +1,147 @@
+package com.box.l10n.mojito.rest.scheduledjob;
+
+import com.box.l10n.mojito.entity.ScheduledJob;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobDTO;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobManager;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobRepository;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobResponse;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobKey;
+import org.quartz.SchedulerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+/**
+ * Webservice for viewing, enabling/disabling and triggering scheduled jobs.
+ *
+ * @author mattwilshire
+ */
+@RestController
+@ConditionalOnProperty(value = "l10n.scheduledJobs.enabled", havingValue = "true")
+public class ScheduledJobWS {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledJobWS.class);
+
+  @Autowired ScheduledJobRepository scheduledJobRepository;
+
+  @Autowired ScheduledJobManager scheduledJobManager;
+
+  private final ResponseEntity<ScheduledJobResponse> notFoundResponse =
+      createResponse(HttpStatus.NOT_FOUND, ScheduledJobResponse.Status.ERROR, "Job doesn't exist");
+
+  @RequestMapping(method = RequestMethod.GET, value = "/api/jobs")
+  @ResponseStatus(HttpStatus.OK)
+  public List<ScheduledJobDTO> getAllJobs() {
+    List<ScheduledJob> scheduledJobs = scheduledJobRepository.findAll();
+    return scheduledJobs.stream().map(ScheduledJobDTO::new).collect(Collectors.toList());
+  }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/api/jobs/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  public ScheduledJobDTO getJob(@PathVariable UUID id) {
+    return scheduledJobRepository
+        .findById(id.toString())
+        .map(ScheduledJobDTO::new)
+        .orElseThrow(
+            () ->
+                new ResponseStatusException(HttpStatus.NOT_FOUND, "Job not found with id: " + id));
+  }
+
+  @RequestMapping(method = RequestMethod.POST, value = "/api/jobs/{id}/trigger")
+  @ResponseStatus(HttpStatus.OK)
+  public ResponseEntity<ScheduledJobResponse> triggerJob(@PathVariable UUID id) {
```

Comment:
What happens if someone tries to trigger a job that is currently disabled?

Reply:
Good catch, it would call the trigger to Quartz, the trigger listener would then check if the job is enabled and then veto the job out, however the response would say job triggered. I fixed this by checking if the job is enabled and only allowing the trigger if the job is enabled. Would it be better to allow the job to be manually triggered even if its disabled ?

Reply:
>fixed this by checking if the job is enabled and only allowing the trigger if the job is enabled
That is preferable to enabling the job and also re-enabling the recurring schedule as a (maybe unintended to the user) side-effect.